executor/executor_xapi.go

```diff
@@ -160,6 +164,9 @@ func (e *XSelectIndexExec) doRequest() error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if len(handles) == 0 {
+		return nil
+	}
 	indexOrder := make(map[int64]int)
 	for i, h := range handles {
 		indexOrder[h] = i
```

Comment:
This may be not necessary. I suggest add an attribute named "keepOrder" in XSelectTableExec to indicate if order should be kept.